File: CoreTransformer/TransitionGroupCollection.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2009-2015 ApexRMS.
'
'************************************************************************************

Imports System.Collections.ObjectModel

''' <summary>
''' Collection of TransitionGroups keyed by TransitionGroupID
''' </summary>
Public Class TransitionGroupCollection
    Inherits KeyedCollection(Of Integer, TransitionGroup)

    ''' <summary>
    ''' Constructor
    ''' </summary>
    ''' <remarks>
    ''' Prevent public constructor
    ''' </remarks>
    Friend Sub New()
        Return
    End Sub

    Protected Overrides Function GetKeyForItem(ByVal item As TransitionGroup) As Integer
        Return item.TransitionGroupId
    End Function

End Class

File: src/Runtime/TransitionGroupCollection.cs
// stsim: A SyncroSim Package for developing state-and-transition simulation models using ST-Sim.
// Copyright © 2007-2021 Apex Resource Management Solutions Ltd. (ApexRMS). All rights reserved.

using System.Collections.ObjectModel;

namespace SyncroSim.STSim
{
    public class TransitionGroupCollection : KeyedCollection<int, TransitionGroup>
    {
        internal TransitionGroupCollection()
        {
            return;
        }

        protected override int GetKeyForItem(TransitionGroup item)
        {
            return item.TransitionGroupId;
        }
    }
}
